docker/client.py

```diff
@@ -140,9 +140,14 @@ def build(self, path=None, tag=None, quiet=False, fileobj=None,
             if self._auth_configs:
                 if headers is None:
                     headers = {}
-                headers['X-Registry-Config'] = auth.encode_full_header(
-                    self._auth_configs
-                )
+                if utils.compare_version('1.19', self._version) >= 0:
```

Comment:
This change was still necessary with 1.7.1. You'll see that the previous code was including a top level key of `configs`. Removing the top level key of `configs` works with 1.7.1. Based on Josh's comment, I'm not sure that was previously necessary. I haven't tried testing to see if removing the `configs` still works against older versions of docker (prior to 1.7).